xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-macro/macroEditor.js

```diff
@@ -358,73 +115,136 @@ define('macroParameterTreeDisplayer', ['jquery', 'l10n!macroEditor'], function($
+      // Some of the child nodes might be hidden so we will activate the first visible tab at the end.
+      tab.add(tabPanel).removeClass('active').toggleClass('hidden', !!childNode.hidden);
+      fillNodeTab(parametersMap, childNode, tab.children().first(), tabPanel);
+    });
+    // Activate the first visible tab.
+    let activeTab = tabs.children().not('.hidden').first();
+    let activeTabPanel = tabPanels.children().not('.hidden').first();
+    activeTab.add(activeTabPanel).addClass('active');
+    toggleMacroParameterGroupVisibility(output);
+    // we remove all occurrences of feature title since we already have the tabs name.
+    output.find('.feature-container .panel-heading').remove();
+    return output;
+  },
+
+  displayGroup = function(parametersMap, groupNode) {
+    return displayFeature(parametersMap, groupNode, groupNode.featureOnly);
+  },
+
+  macroFeatureContainerTemplate =
+    '<div class="feature-container panel panel-default" id="feature-{{featureName}}">' +
+      '<div class="panel-heading">' +
+        '<span class="feature-title">{{featureTitle}}</span>' +
+        '<span class="mandatory"></span>' +
       '</div>' +
-    '</li>',
+      '<div class="panel-body"></div>' +
+    '</div>',
 
-  displayMultipleChoiceGroup = function(groupNode) {
-    var output = $(macroParameterGroupTemplate).addClass('multiple-choice');
-    fillNodeTab(groupNode, output.find('.macro-parameter-group-name'), output.find('.macro-parameter-group-members'));
-    toggleMacroParameterGroupVisibility(output);
+  macroFeatureContentTemplate =
+    '<div class="feature-parameter">' +
+      '<div class="feature-choice">' +
+        '<input type="radio" class="feature-radio" id="feature-radio-{{featureName}}-{{parameterId}}"' +
+      ' name="feature-radio-{{featureName}}" value="{{parameterKey}}" />' +
+        '<label class="feature-choice-name" for="feature-radio-{{featureName}}-{{parameterId}}">' +
+          '{{parameterLabel}}' +
+        '</label>' +
+      '</div>' +
+      '<div class="feature-choice-body"></div>' +
+    '</div>',
+
+  displayFeature = function (parametersMap, featureNode, isFeature) {
+    let name = (isFeature) ? featureNode.featureName : featureNode.name;
+    let output = $(macroFeatureContainerTemplate
+        .replaceAll("{{featureName}}", featureNode.id)
+        .replaceAll("{{featureTitle}}", name));
+    if (featureNode.mandatory) {
+      output.find('.mandatory').text('(' + translations.get('required') + ')');
+      output.addClass('mandatory');
+    }
+    output.find('.panel-body').append(featureNode.children.map(nodeKey => {
+      let paramNode = parametersMap[nodeKey];
+      let nodeOutput = $(macroFeatureContentTemplate
+          .replaceAll("{{featureName}}", featureNode.id)
+          .replaceAll("{{parameterKey}}", nodeKey)
+          .replaceAll("{{parameterId}}", paramNode.id)
+          .replaceAll("{{parameterLabel}}", translations.get('selectFeature', paramNode.name)));
```

Comment:
Same as above, I'd prefer to set the inner text instead of using string replace.